include/uri/UEntity.h

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2023 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef _UENTITY_H_
+#define _UENTITY_H_
+
+#include <algorithm>
+#include <cctype>
+#include <optional>
+#include <string>
+#include "UriFormat.h"
+
+namespace uprotocol::uri {
+
+/**
+ * Data representation of an <b> Software Entity - uE</b><br>
+ * Software entities are distinguished by using a unique name or a unique id along with the specific version of the software.
+ * An Software Entity is a piece of software deployed somewhere on a uDevice.<br>
+ * The Software Entity is used in the source and sink parts of communicating software.<br>
+ * A uE that publishes events is a <b>Service</b> role.<br>
+ * A uE that consumes events is an <b>Application</b> role.
+ */
+class UEntity : public UriFormat {
+public:
+    /**
+     * Constructor for building a UEntity with empty contents.
+     */
+    UEntity() { UEntity::empty(); }
+
+    /**
+     * Create a complete UEntity with all the information so that it can be used in long form UUri serialisation and micro form UUri serialisation.
+     * In the case of missing elements such as name or id, the UEntity will not be marked as resolvable and will not be usable in serialisation formats.
+     * @param name The name of the software such as petapp or body.access.
+     * @param version The software version. If not supplied, the latest version of the service will be used.
+     * @param id A numeric identifier for the software entity which is a one-to-one correspondence with the software name.
+     * @return Returns a complete UEntity with all the information so that it can be used in long form UUri serialisation and micro form UUri serialisation.
+     */
+    static UEntity resolvedFormat(const std::string& name,
+                                  const std::optional<uint8_t> version,
+                                  const std::optional<uint16_t> id) {
+        bool resolved = !isBlank(name) && id.has_value();
+        return UEntity{name, version, id, resolved};
```

Comment:
split to multiline according to coding convention

Reply:
Listing each argument to separate line is intended of function definitions.